File: pinyougou-sellergoods-interface/src/main/java/com/ming/pinyougou/sellergoods/service/BrandService.java
package com.ming.pinyougou.sellergoods.service;

import com.ming.pinyougou.entity.PageResult;
import com.ming.pinyougou.entity.Result;
import com.ming.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**品牌服务
 * @author m969130721@163.com
 * @date 18-10-27 下午2:02
 */
public interface BrandService {

    /**
     * 返回所有列表
     * @return
     */
     List<TbBrand> findAll();

    /**
     * 分页查找
     * @param pageNum
     * @param pageSize
     * @return
     */
     PageResult findPage(Integer pageNum,Integer pageSize);


     PageResult findPage(TbBrand tbBrand,Integer pageNum,Integer pageSize);

    /**
     * 查找一个
     * @param id
     * @return
     */
     TbBrand findByOne(Long id);

    /**
     * 添加
     * @param tbBrand
     * @return
     */
     Result add(TbBrand tbBrand);

    /**
     * 更新
     * @param tbBrand
     * @return
     */
     Result update(TbBrand tbBrand);


    /**
     * 删除
     * @param ids
     * @return
     */
    Result delete(Long[] ids);


    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//service层
app.service("brandService", function ($http) {
    //根据id查找单个品牌
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };
    //分页＋搜索显示数据
    this.search = function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&rows=" + rows, searchEntity);
    };

    //批量删除
    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids);

    };

    this.add = function (entity) {
        return $http.post("../brand/add.do", entity);
    };

    this.update = function (entity) {
        return $http.post("../brand/update.do", entity);
    };
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    };
});